BluetoothApp/Sources/BluetoothApp/main.page.swift

```diff
@@ -0,0 +1,108 @@
+import ScadeKit
+import FusionBluetooth
+
+class Device: EObject {
+    let name: String?
+    let uuid: String
+    var isConnected: Bool
+    
+    init(name: String?, uuid: String, isConnected: Bool) {
+        self.name = name
+        self.uuid = uuid
+        self.isConnected = isConnected
+    }
+}
+
+class MainPageAdapter: SCDLatticePageAdapter {
+
+	let bluetoothManager = BluetoothManager()
+
+	private var deviceArray: [Device] = []
+	private var selectedDevice: Device? = nil
+   
+    override func load(_ path: String) {
+      super.load(path)
+      
+      discoverButton.onClick.append(SCDWidgetsEventHandler{ _ in self.discoverDevicesOrStop()})
+      connectButton.onClick.append(SCDWidgetsEventHandler{ _ in self.connectDisconect()})
+      
+      self.bluetoothManager.requestAuthorization()
+      
+      self.checkState()
```

Comment:
I asked this in the API topic, here, if one wants to change the label text based on the authorisation results, can one just do a request with a authorisation callback without using check